Alexnet with Code (#230)
johko/computer-vision-course: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx

# Alexnet

# Introduction

> In this chapter we will take a look at AlexNet, the network that began the revolution of deep convolutional neural network Architechture that revolutionized the field

Reviewer: "revolution ... that revolutionized" is a bit repetitive. please rephrase.

of computer vision.It was introduced by Alex Krizhevsky, Ilya Sutskever, and Geoffrey Hinton in their 2012 paper [ImageNet Classification with Deep Convolutional Neural Networks](https://proceedings.neurips.cc/paper_files/paper/2012/file/c399862d3b9d6b76c8436e924a68c45b-Paper.pdf) and won the ImageNet Large Scale Visual Recognition Challenge (ILSVRC) in 2012.

> The ImageNet dataset consisted of more than 15 million high resolution images.

Reviewer: This last sentence feels somewhat out of context, or lacking in cohesive ties to the rest of the paragraph.

- The second convolutional layer applies 256 kernels of size 5x5x48.
- Both convolutional layers are followed by ReLU (Rectified Linear Unit) activation and max-pooling layers.

2.Fully Connected Layers:

- After the convolutional layers, there are three fully connected layers.
-The first two fully connected layers have 4096 neurons each, followed by dropout regularization to reduce overfitting.
-The final fully connected layer has 1000 neurons, representing the 1000 object categories in the ImageNet dataset.

3.Softmax Layer:

- The output of the final fully connected layer is passed through a softmax activation function to produce the final class probabilities.

4.Normalization Layers:

- AlexNet also introduced Local Response Normalization (LRN) layers, which helped to improve the generalization performance of the network.

5.ReLU Activation:

- AlexNet was one of the first architectures to effectively utilize the Rectified Linear Unit (ReLU) activation function, which helped to address the vanishing gradient problem and accelerated the training process.

### Key features

> - 'ReLU' is used as an activation function than 'tanh'

Reviewer: ```suggestion
- 'ReLU' is used as an activation function as opposed to 'tanh'
```

    - 'tanh' was the standard activation function which was used back in those days.

>     - The primary advantage than 'ReLU' offered was the fact it reduced training time and also help in helped to alleviating the vanishing gradient problem

Reviewer: ```suggestion
    - The primary advantage than 'ReLU' offered was the fact it reduced training time and also help helped to alleviate the vanishing gradient problem
```

- Usage of Dropout mechanism
    - It is a regularization technique that helps to prevent overfitting in neural networks by randomly dropping out (setting to zero) a proportion of units (neurons) and their connections during the training process.
    - Forces the network to learn more robust representation

> - Data Augmentation techniques such as flipping,jittering,cropping etc were performed.

Reviewer: ```suggestion
- Data Augmentation techniques such as flipping, jittering, cropping, etc. were performed.
```

>     - The authors employed employed extensive data augmentation techniques to artificially increase the size and diversity of the training data.

Reviewer: ```suggestion
    - The authors employed extensive data augmentation techniques to artificially increase the size and diversity of the training data.
```

>     - This helped in not only increasing the dataset size , but also improved generalization and acted as a form of regularization.

Reviewer: ```suggestion
    - This helped in not only increasing the dataset size, but also aided generalization and acted as a form of regularization.
```

### Example implementation using Pytorch

> ```

Reviewer: ```suggestion
```python
```

# Alexnet

# Introduction

> In this chapter we will take a look at AlexNet, the network that began the revolution of deep convolutional neural network Architechture.
> of computer vision.It was introduced by Alex Krizhevsky, Ilya Sutskever, and Geoffrey Hinton in their 2012 paper [ImageNet Classification with Deep Convolutional Neural Networks](https://proceedings.neurips.cc/paper_files/paper/2012/file/c399862d3b9d6b76c8436e924a68c45b-Paper.pdf) and won the ImageNet Large Scale Visual Recognition Challenge (ILSVRC) in 2012.

Reviewer 2: ```suggestion
In this chapter we will take a look at AlexNet, the network that began the revolution of Deep Convolutional Neural Network architectures.
It was introduced by Alex Krizhevsky, Ilya Sutskever, and Geoffrey Hinton in their 2012 paper [ImageNet Classification with Deep Convolutional Neural Networks](https://proceedings.neurips.cc/paper_files/paper/2012/file/c399862d3b9d6b76c8436e924a68c45b-Paper.pdf) and won the ImageNet Large Scale Visual Recognition Challenge (ILSVRC) in 2012.
```

The ImageNet dataset consisted of more than 15 million high resolution images that posed a significant challenge to traditional computer vision algorithms, as it demanded an unprecedented level of visual understanding and classification capabilities.
AlexNet's triumph in the ILSVRC marked a turning point in the field of computer vision, demonstrating the superior performance of deep convolutional neural networks over conventional machine learning techniques.

> ### Architechture of Alexnet

Reviewer 2: an image/diagram of the architecture would be very nice for a better overview

The AlexNet architecture consists of the following layers:

1. Convolutional Layers:
- The input image is processed by the first convolutional layer, which applies 96 kernels of size 11x11x3 with a stride of 4 pixels.
- The second convolutional layer applies 256 kernels of size 5x5x48.
- Both convolutional layers are followed by ReLU (Rectified Linear Unit) activation and max-pooling layers.

> 2.Fully Connected Layers:

Reviewer 2: ```suggestion
2. Fully Connected Layers:
```

- After the convolutional layers, there are three fully connected layers.

> -The first two fully connected layers have 4096 neurons each, followed by dropout regularization to reduce overfitting.
> -The final fully connected layer has 1000 neurons, representing the 1000 object categories in the ImageNet dataset.

Reviewer 2: ```suggestion
- The first two fully connected layers have 4096 neurons each, followed by dropout regularization to reduce overfitting.
- The final fully connected layer has 1000 neurons, representing the 1000 object categories in the ImageNet dataset.
```

> 3.Softmax Layer:

Reviewer 2: ```suggestion
3. Softmax Layer:
```

- The output of the final fully connected layer is passed through a softmax activation function to produce the final class probabilities.

> 4.Normalization Layers:

Reviewer 2: ```suggestion
4. Normalization Layers:
```

- AlexNet also introduced Local Response Normalization (LRN) layers, which helped to improve the generalization performance of the network.

> 5.ReLU Activation:

Reviewer 2: ```suggestion
5. ReLU Activation:
```

- AlexNet was one of the first architectures to effectively utilize the Rectified Linear Unit (ReLU) activation function, which helped to address the vanishing gradient problem and accelerated the training process.

### Key features

- 'ReLU' is used as an activation function as opposed to 'tanh'
    - 'tanh' was the standard activation function which was used back in those days.

>     - The primary advantage than 'ReLU' offered was the fact it reduced training time and also helped to alleviate the vanishing gradient problem

Reviewer 2: ```suggestion
    - The primary advantage 'ReLU' offered, was the fact that it reduced training time and also helped to alleviate the vanishing gradient problem
```

- Usage of Dropout mechanism
    - It is a regularization technique that helps to prevent overfitting in neural networks by randomly dropping out (setting to zero) a proportion of units (neurons) and their connections during the training process.
    - Forces the network to learn more robust representation

> - Data Augmentation techniques such as flipping ,jittering ,cropping, etc. were performed.

Reviewer 2: ```suggestion
- Data Augmentation techniques such as flipping, jittering, cropping, etc. were performed.
```

>     - The authors employed  extensive data augmentation techniques to artificially increase the size and diversity of the training data.

Reviewer 2: ```suggestion
    - The authors employed extensive data augmentation techniques to artificially increase the size and diversity of the training data.
```
